Fix maven groupId completion

matthiasblaesing opened the pull request:
MavenIndexer dropped the caching of the allGroups and rootGroups
properties in the index:

https://github.com/apache/maven-indexer/commit/c231db602b5cea9ed20eba617de5a5a846ef5faa

After the index is loaded there are two options:

a) rebuild the groups list from the index
b) cache the group list after indexing

NetBeans expects the index to be readily be available and thus option
a is not an option (rebuilding the group list runs over the full index
and is in the minute range). As there are only a few entrypoints that
modify the index caching the group list looks like a viable option.

Closes: #4102

mbien reviewed (approved):
thanks for fixing this, I wasn't sure at first about custom caching, I probably would have tried to add it into the index, but i believe this is the most straight forward way of doing this - I like it.

I added a commit which simplifies the code a little to use nio.Files.* utilities. Tested it and everything works.

Should we wait a bit with integration? Maybe maven-indexer releases a hotfix so that we can drop the "inversion" code.

matthiasblaesing commented:
I looked at your change and looks sane to me, adjusted a little and squashed. I intent to merge once this gets green. maven-indexer talks about going jdk 11+, this might or might not be a problem for NetBeans and I want to get back to being able to use master.

I was surprised by the push into this PR - I'd have expected a heads-up before (yes I'll try to remember to uncheck the "allow edits" checkbox).

mbien commented:
I mentioned it in the review comment, going to attach a diff next time.
